packages/time/date.pony

```diff
@@ -16,9 +16,18 @@ class Date
 
   new create(seconds: I64 = 0, nanoseconds: I64 = 0) =>
     """
-    Create a date from a POSIX time.
+    Create a date from a POSIX time. Negative arguments will be changed to zero.
     """
-    @ponyint_gmtime[None](this, seconds, nanoseconds)
+    @ponyint_gmtime[None](this,
+      negative_to_zero(seconds),
+      negative_to_zero(nanoseconds))
+
+  fun negative_to_zero(value: I64): I64 =>
```

Comment:
This method would not be private. To make it private you could start the name with an underscore, like `_negative_to_zero`. Alternatively, you could define it as a lambda in the constructor body since its only used there:
```pony
new create(seconds: I64 = 0, nanoseconds: I64 = 0) =>
  let positive_or_zero = {(n: I64): I64 => if n > 0 then value else 0 end  }
  @ponyint_gmtime[None](this,
     positive_or_zero(seconds),
     positive_or_zero(nanoseconds))
```